### DISCUSSION TITLE
Rectangle pixels are only square in UVTools? Can it be changed to rectangles?

### DISCUSSION THREAD

**Discussion by markoSVK** [sn4k3/UVtools#866] [General]
Hello,
I draw layers of my works (special filters) in 2D and combine them to 3D in UVTools. 
Then print with a resin printer.
Until now I used to use square pixels (22x22 microns) but was thinking about using a 16.8 x 24.8 rectangle pixel printer.
It is 1.482 ratio. In Photoshop I can set a 1.5 ratio of pixels so it is almost perfect, but when open layers in UVtools, pixels are square.
Some elements are only 3-4 pixels wide so the difference between square and rectangle is visible.
Thank you for ideas.

**Comment by sn4k3** [comment]
UVtools shows you the image as it is and as printer will receive it, in other words they are a Bitmap. If you open that bitmap in a program, eg: png image visualizer, you will notice that it display same image as UVtools. This bitmaps are just pixel value information, it does not represent it size nor aware of any measurement/ratio. For example, it goes to (50,50) position of bitmap fetch the value and lit the corresponding pixel. The size of pixel does not matter, bitmap only contain the information to lit or not the pixel in such position. And bitmaps are "size-less" square grid of y/x.
When you use printers with ratios other than 1:1 you will notice layer images are distorced, that's normal to account the odd pixel size. Just remember than 1 square pixel in uvtools represents 1 pixel of the printer in that position, no matter it size.

Pratical examples:

ratio of 50um x 50um: pixel in 0,0 position will form a 50um x 50um square
ratio of 25um x 50um: pixel in 0,0 position will form a 25um x 50um rectangle, if you want to achieve a size of 50x50 you will need to lit two pixels, so: (0,0 and 0,1) on = 50um x 50um, while png visualizer will render a rectangle, your printer will print a square!

So do not worry about UVtools showing as squares, you can keep importing and follow your workflow it will result in the accurate measure after print.

After import your images you can use shift key, select a area and confirm it real size on status bar: ROI (look for mm measure)

![image](https://github.com/sn4k3/UVtools/assets/113281/4f03aefc-43d9-4f61-898a-f9c544f6cd60)
